chain/client/src/types.rs

```diff
@@ -264,6 +266,29 @@ impl Message for Status {
     type Result = Result<StatusResponse, String>;
 }
 
+pub struct GetNetworkInfo {}
+
+impl Message for GetNetworkInfo {
+    type Result = Result<NetworkInfoResponse, String>;
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PeerInfo {
+    pub addr: Option<SocketAddr>,
```

Comment:
Why is `addr` an option?

Reply:
The original network/src/types.rs FullPeerInfo.PeerInfo.addr is of type Option<SocketAddr> so I assumed it's possibly empty.